### ISSUE TITLE
The camera follow demo does not stop following on double tap zoom

### ISSUE THREAD

**Issue by sargunv** [sargunv/maplibre-compose#77]
Normally the demo stops following when any gesture happens, but the double tap zoom gesture doesn't trigger this. Affects both Android and iOS.

**Comment by sargunv** [comment]
compass part fixed in https://github.com/sargunv/maplibre-compose/pull/179

double tap is still outstanding

**Comment by michalgwo** [comment]
I noticed that when the camera follow animation is run, the move reason is programmatic, but when it's not, the move reason is a gesture. It may be because of the execution of `cameraState.animateTo()` every few hundred milliseconds, which overrides the double tap gesture (see: https://github.com/maplibre/maplibre-native/issues/2790). I limited `cameraState.animateTo()` executions to 1 per second and the move reason was correctly detected as a gesture. 

I'm not sure if it's important, but I also noticed that when the camera follow animation is run and I tap on the map without moving it, it detects it as a programmatic camera movement (double tap, detects 2 programmatic camera movements), but when I tap on the map when there is no animation, it doesn't detect any camera movement.

As for differentiation between the developer's and API's programmatic reasons for camera movement, it seems to be a bug in Maplibre, and it was an issue before the fork on Mapbox (see: https://github.com/mapbox/mapbox-gl-native/issues/10674#issuecomment-526181969)

As a workaround for the demo, we can stop the animation on zoom changes, but it's not a real solution, so probably not a good option.

**Comment by sargunv** [comment]
Yup. We're limited by what the SDK provide here. Calling animateTo so frequently doesn't even work properly on JS.

I'm hoping to solve this by moving the gestures implementation fully into this library. We'll need to reimplement gestures anyway for desktop native, so if we make that common code I hope we can solve this and also get better integration with Compose gesture interop APIs.

**Comment by sargunv** [comment]
#870 eliminates the platform compass so this is now solvable on our end. probably an api that allows a "programmatic" camera movement to register as gesture? won't fully work for android/ios until #572 but solving it for desktop/web will solve it for those too eventually.

**Comment by sargunv** [comment]
thinking some more, breaking follow on compass tap is somewhat weird, and the user can already wire that up if they want. double tap zoom is handled by our compose-implemented gestures now on desktop/web, so what's left here is just duplicate of #572
